tensorboard/plugins/scalar/tf_scalar_dashboard/tf-scalar-card.html

```diff
@@ -314,11 +316,13 @@
       },
       _updateDownloadLink() {
         const svgStr = this.$$('tf-line-chart-data-loader').exportAsSvgString();
-        // The SVG code string may include hash characters, such as an
-        // attribute `clipPath="url(#foo)"`. Thus, we base64-encode the
-        // data so that such a hash is not interpreted as a fragment
-        // specifier, truncating the SVG. (See issue #1874.)
-        this.$$('#svgLink').href = `data:image/svg+xml;base64,${btoa(svgStr)}`;
+        const bytes = new Uint8Array(svgStr.length).map((_, i) => svgStr.charCodeAt(i));
```

Comment:
This will silently do the wrong thing if `svgStr` contains code points
outside of Latin-1. This could result in incorrect data or a corrupt
SVG: e.g., an attribute `x="my • tag"` will become `x="my " tag"`, which
is clearly problematic.

Please use `new TextEncoder().encode(svgStr)` instead.

Reply:
Oh wonderful! JW, I guess this is not relevant to graph because it is taking PNG from the canvas?

Reply:
The code in `minimap.ts` is fine because its data comes from an
immediately preceding `atob`, which only emits code points in 0..255.
(Conversely, `btoa` rejects arguments that have code points outside of
Latin-1.) That said, it would be fine to use `TextEncoder` there anyway.